aws/resource_aws_ebs_snapshot_test.go

```diff
@@ -91,58 +129,71 @@ func testAccCheckSnapshotExists(n string, v *ec2.Snapshot) resource.TestCheckFun
 	}
 }
 
-const testAccAwsEbsSnapshotConfig = `
+func testAccAwsEbsSnapshotConfigBasic(rName string) string {
+	return fmt.Sprintf(`
+data "aws_region" "current" {}
+
 resource "aws_ebs_volume" "test" {
-  availability_zone = "us-west-2a"
+  availability_zone = "${data.aws_region.current.name}a"
   size              = 1
 }
 
 resource "aws_ebs_snapshot" "test" {
   volume_id = "${aws_ebs_volume.test.id}"
 
   tags {
-    Name = "testAccAwsEbsSnapshotConfig"
+    Name = "%s"
   }
 }
-`
+`, rName)
+}
+
+func testAccAwsEbsSnapshotConfigWithDescription(rName string) string {
+	return fmt.Sprintf(`
+data "aws_region" "current" {}
 
-const testAccAwsEbsSnapshotConfigWithDescription = `
 resource "aws_ebs_volume" "description_test" {
-	availability_zone = "us-west-2a"
-	size = 1
+  availability_zone = "${data.aws_region.current.name}a"
+  size              = 1
 }
 
 resource "aws_ebs_snapshot" "test" {
-	volume_id = "${aws_ebs_volume.description_test.id}"
-	description = "EBS Snapshot Acceptance Test"
+  volume_id = "${aws_ebs_volume.description_test.id}"
+  description = "%s"
+}
+`, rName)
 }
-`
 
-const testAccAwsEbsSnapshotConfigWithKms = `
+func testAccAwsEbsSnapshotConfigWithKms(rName string) string {
+	return fmt.Sprintf(`
+variable "name" { default = "%s" }
+data "aws_region" "current" {}
+
 resource "aws_kms_key" "test" {
   deletion_window_in_days = 7
 
   tags {
-    Name = "testAccAwsEbsSnapshotConfigWithKms"
+    Name = "${var.name}"
   }
 }
 
 resource "aws_ebs_volume" "test" {
-  availability_zone = "us-west-2a"
+  availability_zone = "${data.aws_region.current.name}a"
```

Comment:
Minor nitpick: Not all regions have/will have AZ `a` available but this is certainly an improvement (`data.aws_availability_zones.current[0].id` is safer)